File: scratch/fix_errors.js
const fs = require('fs');
const path = require('path');

const dirPath = 'src/app/components';
const files = fs.readdirSync(dirPath).filter(f => 
  f.endsWith('BrochureModal.jsx') || 
  f === 'GeneralApplicationModal.jsx' || 
  f === 'CareerApplicationForm.jsx' ||
  f === 'ContactForm.jsx' ||
  f === 'ContactFormModal.jsx'
);

// Helper to get a friendly name from the filename
function getFormName(filename) {
    if (filename === 'ContactForm.jsx') return 'Contact Form';
    if (filename === 'ContactFormModal.jsx') return 'Demo Request';
    if (filename === 'CareerApplicationForm.jsx') return 'Career Application';
    if (filename === 'GeneralApplicationModal.jsx') return 'General Application';
    return filename.replace('BrochureModal.jsx', ' Brochure Request').replace(/([A-Z])/g, ' $1').trim();
}

files.forEach(f => {
  const fullPath = path.join(dirPath, f);
  let content = fs.readFileSync(fullPath, 'utf8');
  const formName = getFormName(f);
  
  // Regex to match our previously implemented specific logic or the old generic one
  const regex = /if\s*\(typeof\s*data\.error\s*===\s*'string'\)\s*\{\s*if\s*\(data\.error\.includes\('mobile'\)\s*\|\|\s*data\.error\.includes\('phone'\)\)\s*\{\s*friendlyError\s*=\s*data\.error\.includes\('associated'\)\s*\?\s*'This phone number is already registered\.'\s*:\s*'Please enter a valid phone number\.';\s*\}\s*else\s*if\s*\(data\.error\.includes\('email'\)\)\s*\{\s*friendlyError\s*=\s*\(data\.error\.includes\('taken'\)\s*\|\|\s*data\.error\.includes\('associated'\)\)\s*\?\s*'This email address is already registered\.'\s*:\s*'Please enter a valid email address\.';\s*\}\s*\}/g;
  
  // Base case (if script hasn't run yet or we want to overwrite)
  const baseRegex = /if\s*\(data\.error\s*&&\s*typeof\s*data\.error\s*===\s*'string'\)\s*\{\s*if\s*\(data\.error\.includes\('mobile'\)\s*\|\|\s*data\.error\.includes\('phone'\)\)\s*\{\s*friendlyError\s*=\s*'Please enter a (proper|valid) phone number\.';\s*\}\s*else\s*if\s*\(data\.error\.includes\('email'\)\)\s*\{\s*friendlyError\s*=\s*'Please enter a valid email address\.';\s*\}(?:\s*else\s*if\s*\(data\.error\.includes\('name'\)\)\s*\{\s*friendlyError\s*=\s*'Please enter your full name\.';\s*\})?\s*else\s*\{\s*friendlyError\s*=\s*data\.error;\s*\}\s*\}/g;

  let replacement = `if (typeof data.error === 'string') {
          const prefix = 'Failed to submit ${formName}: ';
          if (data.error.includes('mobile') || data.error.includes('phone')) {
            friendlyError = prefix + (data.error.includes('associated') ? 'This phone number is already registered.' : 'Please enter a valid phone number.');
          } else if (data.error.includes('email')) {
            friendlyError = prefix + ((data.error.includes('taken') || data.error.includes('associated')) ? 'This email address is already registered.' : 'Please enter a valid email address.');
          } else {
            friendlyError = prefix + data.error;
          }
        }`;

  if (content.includes('friendlyError = data.error || \'Something went wrong. Please try again.\';')) {
      // Overwrite the manual one I did for ContactFormModal
       content = content.replace(/let friendlyError = data\.error \|\| 'Something went wrong\. Please try again\.';.*?\}/s, `let friendlyError = data.error || 'Something went wrong. Please try again.';\n        ${replacement}`);
       console.log(`Updating ${f} (Manual Overwrite)`);
  } else if (regex.test(content)) {
    console.log(`Updating ${f} (Incremental Update)`);
    content = content.replace(regex, replacement);
  } else if (baseRegex.test(content)) {
    console.log(`Updating ${f} (Base match)`);
    content = content.replace(baseRegex, replacement);
  } else {
     // Even more generic match for the brochure modals
     const brochureRegex = /if\s*\(data\.error\s*&&\s*typeof\s*data\.error\s*===\s*'string'\)\s*\{.*?friendlyError\s*=\s*data\.error;.*?\}\s+\}/s;
     if (brochureRegex.test(content)) {
         console.log(`Updating ${f} (Brochure specific match)`);
         content = content.replace(brochureRegex, replacement + '\n      }');
     } else {
        console.log(`Target not found in ${f}`);
     }
  }
  
  fs.writeFileSync(fullPath, content);
});

File: scratch/fix_errors.py
import os

path = 'src/app/components'
files = [f for f in os.listdir(path) if f.endswith('BrochureModal.jsx') or f == 'GeneralApplicationModal.jsx' or f == 'CareerApplicationForm.jsx']

target = """if (data.error && typeof data.error === 'string') {
          if (data.error.includes('mobile') || data.error.includes('phone')) {
            friendlyError = 'Please enter a proper phone number.';
          } else if (data.error.includes('email')) {
            friendlyError = 'Please enter a valid email address.';
          } else if (data.error.includes('name')) {
            friendlyError = 'Please enter your full name.';
          } else {
            friendlyError = data.error;
          }
        }"""

replacement = """if (typeof data.error === 'string') {
          if (data.error.includes('mobile') || data.error.includes('phone')) {
            friendlyError = data.error.includes('associated') ? 'This phone number is already registered.' : 'Please enter a valid phone number.';
          } else if (data.error.includes('email')) {
            friendlyError = (data.error.includes('taken') || data.error.includes('associated')) ? 'This email address is already registered.' : 'Please enter a valid email address.';
          }
        }"""

for f in files:
    full_path = os.path.join(path, f)
    with open(full_path, 'r', encoding='utf-8') as file:
        content = file.read()
    
    if target in content:
        print(f'Updating {f}')
        new_content = content.replace(target, replacement)
        with open(full_path, 'w', encoding='utf-8') as file:
            file.write(new_content)
    else:
        # Try a slightly different variation without name check (some might have it)
        target_no_name = """if (data.error && typeof data.error === 'string') {
          if (data.error.includes('mobile') || data.error.includes('phone')) {
            friendlyError = 'Please enter a proper phone number.';
          } else if (data.error.includes('email')) {
            friendlyError = 'Please enter a valid email address.';
          } else {
            friendlyError = data.error;
          }
        }"""
        if target_no_name in content:
            print(f'Updating {f} (no-name variant)')
            new_content = content.replace(target_no_name, replacement)
            with open(full_path, 'w', encoding='utf-8') as file:
                file.write(new_content)
        else:
            print(f'Target not found in {f}')
